network/peer/peer.go

```diff
@@ -774,17 +781,67 @@ func (p *peer) handle(msg message.InboundMessage) {
 }
 
 func (p *peer) handlePing(msg *p2p.Ping) {
-	p.observeUptimes(msg.Uptime, msg.SubnetUptimes)
+	if msg.Uptime > 100 {
+		p.Log.Debug(malformedMessageLog,
+			zap.Stringer("nodeID", p.id),
+			zap.Stringer("messageOp", message.PingOp),
+			zap.Stringer("subnetID", constants.PrimaryNetworkID),
+			zap.Uint32("uptime", msg.Uptime),
+		)
+		p.StartClose()
+		return
+	}
+	p.observeUptime(constants.PrimaryNetworkID, msg.Uptime)
+
+	for _, subnetUptime := range msg.SubnetUptimes {
+		subnetID, err := ids.ToID(subnetUptime.SubnetId)
+		if err != nil {
+			p.Log.Debug(malformedMessageLog,
+				zap.Stringer("nodeID", p.id),
+				zap.Stringer("messageOp", message.PingOp),
+				zap.String("field", "subnetID"),
+				zap.Error(err),
+			)
+			p.StartClose()
+			return
+		}
+
+		if !p.MySubnets.Contains(subnetID) {
+			p.Log.Debug(malformedMessageLog,
+				zap.Stringer("nodeID", p.id),
+				zap.Stringer("messageOp", message.PingOp),
+				zap.Stringer("subnetID", subnetID),
+				zap.String("reason", "not tracking subnet"),
+			)
+			p.StartClose()
+			return
+		}
+
+		uptime := subnetUptime.Uptime
+		if uptime > 100 {
+			p.Log.Debug(malformedMessageLog,
+				zap.Stringer("nodeID", p.id),
+				zap.Stringer("messageOp", message.PingOp),
+				zap.Stringer("subnetID", subnetID),
+				zap.Uint32("uptime", uptime),
+			)
+			p.StartClose()
+			return
+		}
+		p.observeUptime(subnetID, uptime)
+	}
 
 	pongMessage, err := p.MessageCreator.Pong()
 	if err != nil {
-		p.Log.Error("failed to create message",
-			zap.Stringer("messageOp", message.PongOp),
+		p.Log.Error(failedToCreateMessageLog,
 			zap.Stringer("nodeID", p.id),
+			zap.Stringer("messageOp", message.PongOp),
 			zap.Error(err),
 		)
+		p.StartClose()
```

Comment:
The peer may close us anyways if we'd dropping the ping message... so we can just close now and maintain the pattern that failing to build a message will close the connection.